dotCMS/src/integration-test/java/com/dotcms/rest/BundleResourceTest.java

```diff
@@ -88,7 +109,118 @@ private void publish(FileInputStream createCOntentFileInputStream) throws DotPub
+        //Create new bundle
+        final String bundleId = insertPublishingBundle(adminUser.getUserId(),new Date());
+
+        //Create a Filter since it's needed to generate the bundle
+        createFilter();
+
+        //Create GenerateBundleForm
+        final GenerateBundleForm bundleForm = new GenerateBundleForm.Builder().bundleId(bundleId).build();
+
+        //Call generate endpoint
+        final AsyncResponse asyncResponse = new MockAsyncResponse((arg) -> {
+
+            final Response generateBundleResponse = (Response)arg;
+            assertEquals(Status.OK.getStatusCode(), generateBundleResponse.getStatus());
+            return true;
+        }, arg -> {
+            fail("Error generating bundle");
+            return true;
+        });
+
+        bundleResource.generateBundle(getHttpRequest(),response,asyncResponse,bundleForm);
+        PublisherAPIImpl.class.cast(APILocator.getPublisherAPI()).getFilterDescriptorMap().clear();
+    }
+
+    /**
+     * Method to Test: {@link BundleResource#downloadBundle(HttpServletRequest, HttpServletResponse, String)}
+     * When: Create a bundle and try to download it, but since the tar.gz has not been generated should fail.
+     * Should: return 404 since the file has not been generated
+     */
+    @Test
+    public void test_downloadBundle_fileNotGenerated_return404() throws DotDataException {
+        //Create new bundle
+        final String bundleId = insertPublishingBundle(adminUser.getUserId(),new Date());
+
+        //Call download endpoint
+        final Response responseResource = bundleResource.downloadBundle(getHttpRequest(),response,bundleId);
+
+        Assert.assertEquals(Status.NOT_FOUND.getStatusCode(),responseResource.getStatus());
+    }
+
+    /**
+     * Method to Test: {@link BundleResource#downloadBundle(HttpServletRequest, HttpServletResponse, String)}
+     * When: Create a bundle and try to download it, since the tar.gz has been generated should succeed.
+     * Should: return 200 since the file has been generated
+     */
+    @Test
+    public void test_downloadBundle_success() throws DotDataException {
+        //Create new bundle
+        final String bundleId = insertPublishingBundle(adminUser.getUserId(),new Date());
+
+        //Create a Filter since it's needed to generate the bundle
+        createFilter();
+
+        //Generate bundle file
+        final Bundle bundle = APILocator.getBundleAPI().getBundleById(bundleId);
+        bundle.setOperation(PushPublisherConfig.Operation.PUBLISH.ordinal());
+        APILocator.getBundleAPI().generateTarGzipBundleFile(bundle);
+
+        PublisherAPIImpl.class.cast(APILocator.getPublisherAPI()).getFilterDescriptorMap().clear();
+
+        //Call download endpoint
+        final Response responseResource = bundleResource.downloadBundle(getHttpRequest(),response,bundleId);
+
+        Assert.assertEquals(Status.OK.getStatusCode(),responseResource.getStatus());
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6239861)